var/spack/repos/builtin/packages/jsonnet/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Jsonnet(MakefilePackage):
+    """A data templating language for app and tool developers based on JSON"""
+
+    homepage = "https://jsonnet.org/"
+    git = "https://github.com/google/jsonnet.git"
+    url = "https://github.com/google/jsonnet/archive/refs/tags/v0.18.0.tar.gz"
+
+    maintainers = ["jcpunk"]
+
+    version("master", branch="master")
+    version("0.18.0", sha256sum="85c240c4740f0c788c4d49f9c9c0942f5a2d1c2ae58b2c71068107bc80a3ced4")
+    version("0.17.0", sha256sum="076b52edf888c01097010ad4299e3b2e7a72b60a41abbc65af364af1ed3c8dbe")
+
+    conflicts("%gcc@:5.4.99", when="@0.18.0:")
+
+    variant("python", default=False, description="Provide Python bindings for jsonnet")
+    depends_on("python", type=("build", "run"), when="+python")
```

Comment:
```suggestion
    extends("python", when="+python")
```
This implies `depends_on` but also sets up things like `PYTHONPATH` and allows you to "activate" a package.